.gitignore

```diff
@@ -32,3 +32,6 @@ env.bak/
 venv.bak/
 .vscode/*
 *.DS_Store
+
+# Data generated by tests
+data/
```

Comment:
interesting. I didn't see this before. Do you happen to know which test generated this dir? This is super generic name too. Perhaps renaming it would be better.

Reply:
it's the standard path of the torchvision.datasets lib,
```
$ ls data/
cifar-10-batches-py  cifar-10-python.tar.gz

$ grep -iR cifar tests

tests/optim/test_layerwise_gradient_scaler.py:        train_ds = torchvision.datasets.CIFAR10(root="./data", train=True, download=True, transform=transform)
```

Reply:
I didn't know that we have a test that download a dataset. I though only benchmark runs do that with data being cached. @anupambhatnagar, can we avoid doing this? This is going to make our test repeatedly downloading and can get blacklisted.

If that is changed, then, we don't really need this gitignore.

Reply:
The cifar data is only downloaded once. If the directory is found in the root of the project, downloading is skipped. This is the behavior on devfair and on CircleCI. The attached screenshot shows is from my devfair box and in this  case the files had already been downloaded by the `test_vision_model` test in a prior run.

In the case that CircleCI uses a new container to run the tests, then the dataset will need to be downloaded on that container. Additionally, given the download rates on in the fair cluster we have never seen the test take over our timeout limit of 70 seconds.

Adding /data to .gitignore is a no-op in my opinion.

 ![Screen Shot 2022-05-26 at 5 18 25 PM](https://user-images.githubusercontent.com/91992329/170604192-f0979dd2-fd75-40e2-ac3d-f4c02a698e62.png)

Reply:
We have seen that a web server black list our test machine IPs if they download too frequently. They may treat this as a DDos attack. E.g. If we have N dev branches and 15 test machines for each branch we are testing, the downloading can easily be large, like 15*N or more. It is not timeout of download speed.

We can wait until it happens (if ever) to fix this. In that case, I am fine with the gitignore. I'd suggest a name like test_layerwise_gradient_scaler_data rather than data.

Reply:
Agree on both points :)

Reply:
Since this is a dataroot for common datasets, I'd suggest we use a name that is scoped to that purpose, rather than a particular test that is using the dataset.

Calling this directory `data/` is pretty common practice; but if that is too broad, maybe `cached_datasets/` ?

Reply:
👍

Reply:
Updated, @min-xu-ai please approve?